File: testdata/jennies/builders/envelope_assignment/PHPBuilder/src/Sandbox/DashboardBuilder.php
<?php

namespace Grafana\Foundation\Sandbox;

/**
 * @implements \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\Sandbox\Dashboard>
 */
class DashboardBuilder implements \Grafana\Foundation\Cog\Builder
{
    protected \Grafana\Foundation\Sandbox\Dashboard $internal;

    public function __construct()
    {
    	$this->internal = new \Grafana\Foundation\Sandbox\Dashboard();
    }

    /**
     * Builds the object.
     * @return \Grafana\Foundation\Sandbox\Dashboard
     */
    public function build()
    {
        return $this->internal;
    }

    public function withVariable(string $name, string $value): static
    {
        $this->internal->variables[] = new \Grafana\Foundation\Sandbox\Variable(
            name: $name,
            value: $value,
        );
    
        return $this;
    }

}

File: testdata/jennies/builders/envelope_assignment/CSharpBuilder/src/Grafana/Foundation/Sandbox/DashboardBuilder.cs
namespace Grafana.Foundation.Sandbox;


public class DashboardBuilder : Cog.IBuilder<Dashboard>
{
    protected readonly Dashboard @internal;

    public DashboardBuilder()
    {
        this.@internal = new Dashboard();
    }

    public DashboardBuilder WithVariable(string name,string value)
    {
        if (this.@internal.Variables == null)
        {
            this.@internal.Variables = new List<Variable>();
        }
    Variable variable = new Variable();
        variable.Name = name;
        variable.Value = value;
        this.@internal.Variables.Add(variable);
        return this;
    }

    public Dashboard Build()
    {
        return this.@internal;
    }
}
